### PR TITLE
Laravel view models command

### PR DESCRIPTION
**Pull request by thewebartisan7**
Added new command for create ViewModels class based on Spatie's package laravel-view-models.

### CODE REVIEW

**Comment by thewebartisan7** [comment]
I have no idea why test failed.

The command `"beyond:make:vm Admin/User/UserViewModel"` should create file in `"base_path() . '/src/App/Admin/User/ViewModels/UserViewModel.php'`

@regnerisch @alexgaal

**Comment by alexgaal** [comment]
You can run `./vendor/bin/pest` locally to test your implementation. I just cloned your repository and ran the tests which are failing locally as well.

**Comment by thewebartisan7** [comment]
> You can run `./vendor/bin/pest` locally to test your implementation. I just cloned your repository and ran the tests which are failing locally as well.

Yes I see locally doing the same.. But I still have no idea why.

I make test similar to others, is required something else? Because test seem to file in "toBeFile()" means it doesn't create it.

I did something similar like for MakeControllerCommandTest:

```php
<?php

namespace Tests\Commands;

test('can make controller', function () {
    $this->artisan('beyond:make:controller Admin/User/UserController');

    expect(base_path() . '/src/App/Admin/User/Controllers/UserController.php')
        ->toBeFile()
        ->toMatchNamespaceAndClassName()
        ->toPlaceholdersBeReplaced();
});
```

```php
<?php

namespace Tests\Commands;

test('can make view model', function () {
    $this->artisan('beyond:make:vm Admin/User/UserViewModel');

    expect(base_path() . '/src/App/Admin/User/ViewModels/UserViewModel.php')
        ->toBeFile()
        ->toMatchNamespaceAndClassName()
        ->toPlaceholdersBeReplaced();
});
```

**Comment by alexgaal** [comment]
Ah, due to you are testing if there is a required dependency installed (see `BaseCommand.php L33`) this test is failing since it's not installed. `composer require --dev spatie/laravel-view-models` should fix that issue.

**Comment by thewebartisan7** [comment]
> Ah, due to you are testing if there is a required dependency installed (see `BaseCommand.php L33`) this test is failing since it's not installed. `composer require --dev spatie/laravel-view-models` should fix that issue.

hmm... but I have added in composer.json  `"spatie/laravel-view-models": "^1.5"` in require-dev. So it should install it..

What am I missing?

**Comment by thewebartisan7** [comment]
Ah ok I see now 

    $composer = $this->app->make(ComposerContract::class);

    $composer->setPackages(['spatie/data-transfer-object']);

**Comment by alexgaal** [comment]
Thank you for clarifying missing information when someone else wants to contribute. We'll improve that.

File: src/LaravelBeyondServiceProvider.php [hunk 1]
```diff
@@ -40,6 +41,7 @@ public function boot()
                 MakeCommandCommand::class,
                 MakeControllerCommand::class,
                 MakeDataTransferObjectCommand::class,
+                MakeViewModelCommand::class,
```
Comment (alexgaal) [inline src/LaravelBeyondServiceProvider.php:44]: Please append alphabetically. Should be after `MakeServiceProviderCommand`.

**Review by alexgaal — CHANGES_REQUESTED** [review]

**Review by alexgaal — APPROVED** [review]

**Comment by alexgaal** [comment]
GitHub tells me your commits are not signed: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

**Comment by thewebartisan7** [comment]
> GitHub tells me your commits are not signed: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

Yes right.. I am from another laptop right now.. 
In case you want verified commit, I can do that this evening or tomorrow.. I suppose that I need to open new PR.

**Comment by thewebartisan7** [comment]
> GitHub tells me your commits are not signed: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

I was with another laptop. After setup verification, I messed up things trying to sign commits, so I create a new PR here https://github.com/regnerisch/laravel-beyond/pull/61

Sorry.

**Pull request closed without merging**